server/src/domain/asset/asset.service.ts

```diff
@@ -476,11 +517,20 @@ export class AssetService {
     }
 
     // Replace the parent of the stack children with a new asset
-    if (asset.stack && asset.stack.length != 0) {
-      const stackIds = asset.stack.map((a) => a.id);
-      const newParentId = stackIds[0];
-      await this.assetRepository.updateAll(stackIds.slice(1), { stackParentId: newParentId });
-      await this.assetRepository.updateAll([newParentId], { stackParentId: null });
+    if (asset.stack?.primaryAssetId === id) {
+      const stackIds = asset.stack.assets.map((a) => a.id);
+      if (stackIds.length > 2) {
+        console.log('Updating stack, old primary asset', id);
+        const primaryAssetId = stackIds.find((a) => a !== id)!;
```

Comment:
Shouldn't this be equal?

Reply:
So this code is within an if block that only executes if the asset being deleted is the primary asset of a stack. In this case, `id` is the id of the primary asset in the stack. We then check the stack has more than 2 assets (if it has 2, it'll only have 1 after this asset is deleted), and if that check passes, we find a new primary asset. This `stackIds.find` searches for an asset that has an id other than the current `id`, which will then be the new primaryAssetId. I have renamed some of the variables to make what is going on a little clearer :smile: